client/src/pages/members/Members.js

```diff
@@ -0,0 +1,227 @@
+  function enterEditingMode() {
+    setAddIconClassName(addIconDefault);
+    setRemoveIconClassName(removeIconDefault);
+    setEnterEditingIconClassName(enterEditingIconDefault + ' hide');
+    setExitEditingIconClassName(exitEditingIconDefault);
+    setEditingMode(true);
+  }
+
+  /** Handles clicking the checkmark icon to exit member editing mode */
+  function exitEditingMode() {
+    setAddIconClassName(addIconDefault + ' hide');
+    setRemoveIconClassName(removeIconDefault + ' hide');
+    setEnterEditingIconClassName(enterEditingIconDefault);
+    setExitEditingIconClassName(exitEditingIconDefault + ' hide');
+    setEditingMode(false);
+    setIsAddingMember(false);
+  }
+
+  /** Handles clicking the plus icon to add a new member */
+  function addMember() {
+    if (!isAddingMember) {
+      setIsAddingMember(true);
+    } else {
+      confirmAddMember();
+    }
+  }
+
+  async function confirmAddMember() {
+      var nameTextbox = document.getElementById('addMemberNameTB');
+      var emailTextbox = document.getElementById('addMemberEmailTB');
+      var roleTextbox = document.getElementById('addMemberRoleTB');
+      var privilegeDropdown = document.getElementById('privilegeDropdown');
+
+      if (nameTextbox.value !== '' && emailTextbox.value !== ''
+          && roleTextbox.value !== '' && privilegeDropdown !== '') {
+        try {
+          const response = await axios({
+            method: 'post',
+            url: 'http://localhost:8000/addUser',
+            data: {
+              isAdmin: (privilegeDropdown.value === 'Admin' ? true : false),
+              username: nameTextbox.value
+            }
+          });
+          console.log(response);
+        } catch(error) {
+            console.log(error);
+        }
+        fetchMembers();
+        setIsAddingMember(false);
+      }
+  }
+
+  async function removeMembers() {
+    //Go through each checkbox and determine which users should be deleted
+    var userIdsToDelete = [];
+    members.forEach((member) => {
+      var id = member.id;
+      var checkbox = document.getElementById('checkbox' + id);
+      if (checkbox !== null && checkbox.checked) {
+        userIdsToDelete.push(id);
+      }
+    })
+    //Make backend DELETE request
+    if (userIdsToDelete != null && userIdsToDelete.length > 0) {
+      try {
+        const response = await axios({
+          method: 'delete',
+          url: 'http://localhost:8000/deleteUser',
+          data: {
+            listOfIds: userIdsToDelete
+          }
+        });
+      } catch(error) {
+          console.log(error);
+      }
+      fetchMembers();
+    }
+  }
+
+  /** Handles clicking the Select All checkbox */
+  function selectAllCheckbox() {
+    const selectAllCb = document.getElementById('selectAllCheckbox');
+    members.forEach((member) => {
+      var id = member.id;
+      var checkbox = document.getElementById('checkbox' + id);
+      checkbox.checked = selectAllCb.checked;
```

Comment:
Because of encapsulation/abstraction, you should not be directly modifying a TableRow's checkbox like this. Instead, have 'checked' as a property of a TableRow and changed the TableRow's props

Reply:
I recommend each object within members, have a `member.checked` value keep track of if the row is checked. Then pass that as the prop of a TableRow. You should use React state as the central source of truth to track of state rather than individual checkboxes